graph-builder/src/registry.rs

```diff
@@ -42,6 +43,21 @@ impl Into<cincinnati::Release> for Release {
     }
 }
 
+/// Module for the release cache
+pub mod cache {
+    use crate::registry::Release;
+    use futures_locks_pre::RwLock as FuturesRwLock;
+    use std::collections::HashMap;
+
+    /// The cache type to hold the release cache
+    pub type Cache = FuturesRwLock<HashMap<u64, Option<Release>>>;
+
+    /// Instantiates a new release cache
+    pub fn new() -> Cache {
+        FuturesRwLock::new(HashMap::new())
```

Comment:
I'm a bit worried about cache growth - do we know if its significant? Should it be periodically purged? Should we use some kind of LRU here?

Reply:
I think the discussion isn't in the scope of this PR, as the cache semantics aren't changed here, but only how the cache is accessed.

Reply:
Agree - lets watch the memory consumption on stage and act if it keeps growing

Reply:
yep, we no longer fit in 256Mb on stage. Not sure if its due to HashMap or concurrency though